### PR TITLE
Use for...of loops when possible

### PR DESCRIPTION
**Pull request by XhmikosR**
For...of is more consistent and this way we don't need to explicitly check for own properties.

That being said, I hit a case where an error was thrown so I had to add a fallback, which means we could probably improve that part: https://github.com/svg-sprite/svg-sprite/pull/591/files#diff-79d35363069d02f38101ba80dad33c3dd32e39f32af88131c3365a3001a14b47R186

```js
for (const [extension, value] of Object.entries(this.config.render || {})) {
    tasks.push(createResourceTask(value, this.data, this._spriter, extension));
}
```

Also, unfortunately, the current tests don't cover everything yet.

TODO:

- [ ] Confirm all changes are good

### CODE REVIEW

File: lib/svg-sprite.js
Comment (XhmikosR) [inline lib/svg-sprite.js:378]: I added this for consistency, but happy to split/drop the change.

File: lib/svg-sprite/mode/base.js
Comment (XhmikosR) [inline lib/svg-sprite/mode/base.js:185]: This is the fallback I needed to add. Maybe we could make this branch more explicit with an if?

File: lib/svg-sprite/mode/symbol.js
Comment (XhmikosR) [inline lib/svg-sprite/mode/symbol.js:137]: There's one `Array.prototype.forEach.call` here but I couldn't get rid of it last time I tried, and we don't have tests for this mode yet.

File: lib/svg-sprite/shape.js
Comment (XhmikosR) [inline lib/svg-sprite/shape.js:736]: We could/should move these kind of long expressions to variables before the loop

File: lib/svg-sprite/mode/standalone.js
Comment (XhmikosR) [inline lib/svg-sprite/mode/standalone.js:68]: not sure about this patch, needs more eyes.

File: lib/svg-sprite/mode/standalone.js
Comment (Kreeg) [inline lib/svg-sprite/mode/standalone.js:95]: Maybe this changes should be in another PR?

Comment (XhmikosR) [reply lib/svg-sprite/mode/standalone.js:95]: I didn't change that, make sure you have whitespace disabled

**Comment by XhmikosR** [comment]
@Kreeg I'd like to land this if possible, could you review it please? I'll manually squash any patches before the merge.

**Comment by XhmikosR** [comment]
Oh, and if you could fix the test failure, that would help!

I'm mostly not sure about the CLI changes since those are not covered by tests at all.

**Comment by XhmikosR** [comment]
@Kreeg can you review this please? Tests pass now, so the thing that's not really tested is only the CLI.

File: lib/svg-sprite/mode/symbol.js
Comment (XhmikosR) [inline lib/svg-sprite/mode/symbol.js:137]: This needs a second pair of eyes

**Comment by Kreeg** [comment]
@XhmikosR sorry, I missed the notifications. I'll review this later this week.

File: bin/svg-sprite.js
Comment (Kreeg) [inline bin/svg-sprite.js:224]: This should also moved to constants in future

File: lib/svg-sprite.js
Comment (Kreeg) [inline lib/svg-sprite.js:306]: This is also already an array

File: lib/svg-sprite.js
Comment (Kreeg) [inline lib/svg-sprite.js:398]: The sorting here is changed. But also I'm not sure of sorting here is needed

File: bin/svg-sprite.js
Comment (XhmikosR) [reply bin/svg-sprite.js:224]: Yeah, this and the other TODO would be nice to switch to for of, but I couldn't make it work myself. Feel free to push to this branch.

**Comment by XhmikosR** [comment]
@Kreeg feel free to push to this branch if you have any fixes. It'd help land this. Remember that I started the branch a long time ago, and since then quite a few things have changed,.

File: lib/svg-sprite.js
Comment (XhmikosR) [reply lib/svg-sprite.js:398]: Not sure either now that you mention. We could remove sorting. Feel free to push to this branch.

File: lib/svg-sprite/mode/symbol.js
Comment (Kreeg) [reply lib/svg-sprite/mode/symbol.js:137]: LGTM

**Comment by XhmikosR** [comment]
@Kreeg can we move some of the non-related patches to another branch? I personally have moved some of mine to https://github.com/svg-sprite/svg-sprite/compare/for-of...xmr/dev?w=1 so that we keep this branch only for the for...of changes.

**Comment by XhmikosR** [comment]
I finished moving our non-related patches to the [xmr/dev](https://github.com/svg-sprite/svg-sprite/compare/for-of...xmr/dev?w=1) branch so that we land this one squashed and the rest as individual patches :)

File: lib/svg-sprite.js
Comment (XhmikosR) [reply lib/svg-sprite.js:398]: Alternatively, we could try using something like:

```js
for (const [file, size] of Object.fromEntries(Object.entries(sizes).sort())) {
    this.verbose('Created %s: %s', file, size);
}
```

**Comment by XhmikosR** [comment]
@Kreeg LGTY so that we land this?

**Comment by Kreeg** [comment]
@XhmikosR wait pls, this is a big PR, I checked only 60%

**Review by Kreeg — APPROVED** [review]

**Comment by Kreeg** [comment]
@XhmikosR pls check my latest changes and we can go

**Comment by XhmikosR** [comment]
@Kreeg thanks! I moved the logic into a variable. Assuming sorting is OK, we should be good to go. Worst case, we break the log sorting (which we could probably just drop anyway).

**Pull request merged**
